packages/cohere/src/convert-to-cohere-chat-prompt.ts

```diff
@@ -1,13 +1,37 @@
 import {
+  LanguageModelV2CallWarning,
   LanguageModelV2Prompt,
   UnsupportedFunctionalityError,
 } from '@ai-sdk/provider';
 import { CohereAssistantMessage, CohereChatPrompt } from './cohere-chat-prompt';
 
-export function convertToCohereChatPrompt(
-  prompt: LanguageModelV2Prompt,
-): CohereChatPrompt {
+function isSupportedTextMediaType(mediaType?: string): boolean {
+  if (!mediaType) return false;
+
+  switch (mediaType) {
+    // Basic text formats that can be processed as documents
+    case 'text/plain':
+    case 'text/markdown':
+    case 'text/csv':
+    case 'application/json':
+      return true;
+
+    default:
+      // Also support any other text/* media types
+      return mediaType.startsWith('text/');
+  }
```

Comment:
can be simplified to basic ||

return mediaType?.startsWith("text/") || mediaType === 'application/json'